packages/x-date-pickers/src/internals/components/PickersProvider.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { PickerValidDate } from '../../models';
+import { PickersInputLocaleText } from '../../locales';
+import { LocalizationProvider } from '../../LocalizationProvider';
+
+export const PickersFieldContext = React.createContext<PickersFieldContextValue | null>(null);
+
+/**
+ * Provides the context for the various parts of a picker component:
+ * - fieldContextValue: the context for the field
+ * - localizationProvider: the translations passed through the props and through a parent LocalizationProvider.
+ *
+ * @ignore - do not document.
+ */
+export function PickersProvider<TDate extends PickerValidDate>(
+  props: PickersFieldProviderProps<TDate>,
+) {
+  const { fieldContextValue, localeText, children } = props;
+
+  return (
+    <PickersFieldContext.Provider value={fieldContextValue}>
+      <LocalizationProvider localeText={localeText}>{children}</LocalizationProvider>
+    </PickersFieldContext.Provider>
+  );
+}
+
+interface PickersFieldProviderProps<TDate extends PickerValidDate> {
+  fieldContextValue: PickersFieldContextValue;
+  localeText: PickersInputLocaleText<TDate> | undefined;
+  children: React.ReactNode;
+}
+
+export interface PickersFieldContextValue {
+  /**
+   * Open the view if they are closed, close them otherwise.
```

Comment:
How would you name it?
`onToggle` could be for something else that opening / closing, I don't think it's descriptive enough.

Otherwise we go back to `onOpen` / `onClose`
Or `onToggleOpening`?

Reply:
Have you considered going lower-level and providing the `onOpen` and `onClose` callbacks?
This would let users decide on the desired behavior and possibly create separate open and close triggers. 🤔

Reply:
Is there a problem with exposing the `open` flag? 🤔 
I suspect users would eventually want it to have some custom behavior depending on the `open` state. 🙈

Reply:
No, no problem exposing it
It was just to make sure that we agreed to expose it

So we can expose:

- `onOpen`
- `onClose`
- `open`

Reply:
Change done :+1: